ClientTests/FeatureFlagsUserPrefsMigrationUtilityTests.swift

```diff
@@ -0,0 +1,200 @@
+        profile.prefs.setString(UserFeaturePreference.disabled.rawValue,
+                                forKey: legacyFlags.CustomWallpaper + userPrefsSuffix)
+
+        profile.prefs.setString(UserFeaturePreference.enabled.rawValue,
+                                forKey: legacyFlags.ASPocketStories + userPrefsSuffix)
+        profile.prefs.setString(UserFeaturePreference.enabled.rawValue,
+                                forKey: legacyFlags.HistoryHighlightsSection + userPrefsSuffix)
+        profile.prefs.setString(UserFeaturePreference.enabled.rawValue,
+                                forKey: legacyFlags.TopSiteSection + userPrefsSuffix)
+
+        profile.prefs.setString(StartAtHomeSetting.always.rawValue,
+                                forKey: legacyFlags.StartAtHome + userPrefsSuffix)
+
+        // Migrate
+        FeatureFlagUserPrefsMigrationUtility(with: profile).attemptMigration()
+
+        // Verify that the expected settings have the expected settings.
+        guard let pocketSetting = profile.prefs.boolForKey(newFlags.ASPocketStories),
+              let recentlySavedSetting = profile.prefs.boolForKey(newFlags.RecentlySavedSection),
+              let customWallpaperSetting = profile.prefs.boolForKey(newFlags.CustomWallpaper),
+              let historyHighlightsSetting = profile.prefs.boolForKey(newFlags.HistoryHighlightsSection),
+              let topSiteSetting = profile.prefs.boolForKey(newFlags.TopSiteSection),
+              let startAtHomeSetting = profile.prefs.stringForKey(newFlags.StartAtHome),
+              let migrationCheckFlag = profile.prefs.boolForKey(legacyFlags.MigrationCheck)
+        else {
+            XCTFail("Something went wrong finding a value after migration.")
+            return
+        }
+
+        XCTAssertEqual(startAtHomeSetting, StartAtHomeSetting.always.rawValue)
+        XCTAssertFalse(recentlySavedSetting)
+        XCTAssertFalse(customWallpaperSetting)
+        XCTAssertTrue(historyHighlightsSetting)
+        XCTAssertTrue(topSiteSetting)
+        XCTAssertTrue(pocketSetting)
+        XCTAssertNil(profile.prefs.boolForKey(newFlags.SponsoredShortcuts))
+        XCTAssertNil(profile.prefs.boolForKey(newFlags.PullToRefresh))
+        XCTAssertNil(profile.prefs.boolForKey(newFlags.JumpBackInSection))
+        XCTAssertNil(profile.prefs.boolForKey(newFlags.InactiveTabs))
+        XCTAssertNil(profile.prefs.boolForKey(newFlags.SponsoredShortcuts))
+        XCTAssertNil(profile.prefs.boolForKey(newFlags.TabTrayGroups))
+        XCTAssertNil(profile.prefs.boolForKey(newFlags.HistoryGroups))
+        XCTAssertTrue(migrationCheckFlag)
+    }
+
+    // MARK: - Helper methods
+    private func verifyNilStateForKeys() {
+        keyDictionary.forEach { oldKey, newKey in
+            let oldKey = oldKey + "UserPreferences"
+            let newKey = newKey
+            XCTAssertNil(profile.prefs.stringForKey(oldKey))
+            XCTAssertNil(profile.prefs.boolForKey(newKey))
```

Comment:
`file: StaticString = #filePath, line: UInt = #line` could be passed as argument and then passed to the assert, so in case this fails we would know it's failing at which line in which test

Reply:
ooo. cool. Will do that. Ty!